src/legacy/server/index_patterns/routes.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { first } from 'rxjs/operators';
+import { schema } from '@kbn/config-schema';
+import { InternalCoreSetup, KibanaRequest } from 'kibana/server';
+import { IndexPatternsService } from './service';
+
+export function registerRoutes(core: InternalCoreSetup) {
+  const getIndexPatternsService = async (request: KibanaRequest): Promise<IndexPatternsService> => {
+    const client = await core.elasticsearch.dataClient$.pipe(first()).toPromise();
+    const callCluster = (endpoint: any, params: any, options?: any) =>
+      client.asScoped(request).callAsCurrentUser(endpoint, params, options);
+    return new Promise(resolve => resolve(new IndexPatternsService(callCluster)));
+  };
+
+  const router = core.http.createRouter('/api/index_patterns');
+  router.get(
+    {
+      path: '/_fields_for_wildcard',
+      validate: {
+        query: schema.object({
+          pattern: schema.string(),
+          meta_fields: schema.string({ defaultValue: '[]' }),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const indexPatterns = await getIndexPatternsService(request);
+      const { pattern, meta_fields: metaFields } = request.query;
+
+      let parsedFields = [];
+      try {
+        parsedFields = JSON.parse(metaFields);
+      } catch (error) {
+        return response.badRequest();
+      }
+
+      const fields = await indexPatterns.getFieldsForWildcard({
```

Comment:
No `try/catch` here because there wasn't an API test case for not found result in `_fields_for_wildcard`. 

Is it intended?

Reply:
I would add the test and the try/catch to match what you had to do for the other one for the tests to pass. I'm pretty sure it was just an oversight, not having the test.

Reply:
I added the test to check if `_fields_for_wildcard` returns 404.